### DISCUSSION TITLE
Localization of Fluent.Ribbon texts

### DISCUSSION THREAD

**Discussion by avikramaditya** [fluentribbon/Fluent.Ribbon#1225] [Q&A] [answered]
I see that there are standard set of language localizations under RibbonLocalization, for example English.cs, Spanish.cs and so on. There are 2 questions, one is what is the correct way to add locale like en-US or en-GB,.... and other is how to override the translations of existing strings mainly where fallback text is used.

**Comment by batzen** [comment, marked as answer]
If you want to add a localization for "en-US" or "en-GB" you can just create new classes inheriting from `RibbonLocalizationBase` and add the attribute `[RibbonLocalization("English (US)", "en-US")]` or `[RibbonLocalization("English (GB)", "en-GB")]`.

I don't fully understand the second part of your question.
If you want to add text for missing translations you can either create a PR here, or create your own localization classes where you provide/override them.

**Comment by avikramaditya** [comment]
I used slightly different approach where I set RibbonLocalization.Current.Localization = new MyLocalization();
MyLocalization inherits from RibbonLocalizationBase and I override all the text properties and the value for each text property comes from resource file(resx), so based on the CultureInfo.Culture, .net framework loads the locale specific resource dlls and present the translated text accordingly if available.
